gems/aws-sdk-s3/lib/aws-sdk-s3/plugins/md5s.rb

```diff
@@ -6,81 +6,26 @@ module Aws
   module S3
     module Plugins
       # @api private
-      # This plugin is effectively deprecated in favor of modeled
+      # This plugin is deprecated in favor of modeled
       # httpChecksumRequired traits.
       class Md5s < Seahorse::Client::Plugin
-        # These operations allow Content MD5 but are not required by
-        # httpChecksumRequired. This list should not grow.
-        OPTIONAL_OPERATIONS = [
-          :put_object,
-          :upload_part
-        ]
-
-        # @api private
-        class Handler < Seahorse::Client::Handler
-
-          CHUNK_SIZE = 1 * 1024 * 1024 # one MB
-
-          def call(context)
-            if !context[:checksum_algorithms] && # skip in favor of flexible checksum
-               !context[:s3_express_endpoint] # s3 express endpoints do not support md5
-              body = context.http_request.body
-              if body.respond_to?(:size) && body.size > 0
-                context.http_request.headers['Content-Md5'] ||= md5(body)
-              end
-            end
-            @handler.call(context)
-          end
-
-          private
-
-          # @param [File, Tempfile, IO#read, String] value
-          # @return [String<MD5>]
-          def md5(value)
-            if (File === value || Tempfile === value) && !value.path.nil? && File.exist?(value.path)
-              OpenSSL::Digest::MD5.file(value).base64digest
-            elsif value.respond_to?(:read)
-              md5 = OpenSSL::Digest::MD5.new
-              update_in_chunks(md5, value)
-              md5.base64digest
+        option(:compute_checksums,
+               default: true,
+               doc_type: 'Boolean',
+               docstring: <<~DOCS)
+                 This option is deprecated. Please use `:request_checksum_calculation`
+                 instead. When `true`, `request_checksum_calculation` is set to `when_supported`,
+                 and if `false` it is set to `when_required`.
+               DOCS
+
+        def after_initialize(client)
+          client.config.request_checksum_calculation =
```

Comment:
I see why we need to do this for the legacy option.... but I am pretty concerned about the customer behavior from this.  Doing this effectively means that `request_checksum_calculation` does not work at all for S3 - that is, its value is ignored no matter where it is set (client, ENV, shared config).  I think thats pretty confusing.

Reply:
Yeah, agreed. I think `client.config.request_checksum_calculation = 'when_required' unless client.config.compute_checksums` satisfies the concern?

User doesn’t set compute checksums:
* This value defaults to true
* User can freely set new config to required or optional and it will be honored.

User set compute checksums to false:
* Setting new config to when_required basically does the same thing
* Setting to when_optional would maybe be confusing because it would be overwritten, but they have explicitly already chosen a conflicting option of compute checksums false.

I can update the documentation but the "false" case and "when_optional" is probably a less likely use case. However computing checksums in general would make sense to override the when optional anyway.